### PR TITLE
cmake: Fix SDK-less builds

### PR DESCRIPTION
**Pull request by SebastianBoe**
This change changes the semantics of the environment variable
ZEPHYR_SDK_INSTALL_DIR to allow the use of 3rd party toolchains
alongside the Zephyr SDK's host tools.

Specifically, setting ZEPHYR_SDK_INSTALL_DIR now indicates that the
Zephyr SDK host tools are to be used. But not necessarily that the
Zephyr SDK's toolchain is to be used.

The documentation is also changed to explain this behaviour.

Signed-off-by: Sebastian Boe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

File: doc/getting_started/getting_started.rst [hunk 1]
```diff
@@ -116,19 +116,27 @@ The ELF binaries generated by the build system are named by default
...
+Building without the Zephyr SDK
+===============================
 
 The Zephyr SDK is provided for convenience and ease of use. It provides
 cross-compilers for all ports supported by the Zephyr OS and does not require
 any extra flags when building applications or running tests.
 
-If you have a custom cross-compiler or if you wish to use a vendor provided SDK,
-follow the steps below to build with any custom or 3rd party cross-compilers:
+In addition to cross-compilers the Zephyr SDK also provides prebuilt
```
Comment (dbkinder) [inline doc/getting_started/getting_started.rst]: add a comma:

    cross-compilers, the

File: doc/getting_started/getting_started.rst [hunk 2]
```diff
@@ -116,19 +116,27 @@ The ELF binaries generated by the build system are named by default
...
+host tools. To use the SDK host tools alongside a custom or 3rd party
+cross-compiler keep the ZEPHYR_SDK_INSTALL_DIR environment variable
```
Comment (dbkinder) [inline doc/getting_started/getting_started.rst]: add a comma:

    cross-compiler, keep

File: doc/getting_started/getting_started.rst [hunk 3]
```diff
@@ -116,19 +116,27 @@ The ELF binaries generated by the build system are named by default
...
+set to the Zephyr SDK installation directory.
+
+To build without the Zephyr SDK's prebuilt host tools the
```
Comment (dbkinder) [inline doc/getting_started/getting_started.rst]: add a comma:

    host tools, the

File: doc/getting_started/getting_started.rst [hunk 4]
```diff
@@ -152,6 +160,20 @@ follow the steps below to build with any custom or 3rd party cross-compilers:
...
+installing a cross compiler.
+
+.. _third_party_x_compilers:
+
+Using Custom and 3rd Party Cross Compilers
+==========================================
+
+To use a 3rd party cross compiler that is not provided by the Zephyr
+SDK, follow the steps below. It is possible to use a 3rd party cross
+compiler and still use the Zephyr SDK's host tools, see :ref:`the
```
Comment (dbkinder) [inline doc/getting_started/getting_started.rst]: change the comma to a period and start a new sentence:

    SDK's host tools.  See

**Review by dbkinder — DISMISSED** [review]
a few punctuation issues, otherwise looks great

File: doc/getting_started/getting_started.rst (hunk 1 above)
Comment (SebastianBoe) [reply doc/getting_started/getting_started.rst]: Done

File: doc/getting_started/getting_started.rst (hunk 2 above)
Comment (SebastianBoe) [reply doc/getting_started/getting_started.rst]: Done

File: doc/getting_started/getting_started.rst (hunk 3 above)
Comment (SebastianBoe) [reply doc/getting_started/getting_started.rst]: Done

File: doc/getting_started/getting_started.rst (hunk 4 above)
Comment (SebastianBoe) [reply doc/getting_started/getting_started.rst]: Done

**Comment by SebastianBoe** [comment]
@dbkinder : Thanks for the review.

**Comment by SebastianBoe** [comment]
Can't see a "dismiss review" button for some reason, but @dbkinder , all requested changes are fixed.

**Review by nashif — APPROVED** [review]

**Pull request merged**